### ISSUE TITLE
Unable to exclude directory when name contains multiple underscores

### ISSUE THREAD

**Issue by corydeppen** [palantir/tslint#2975]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.4.1
- __Running TSLint via__: CLI

#### Actual behavior

Given a `__test_data__` directory in the root of my project, I run the following command:

```sh
$ ./node_modules/.bin/tslint --project tsconfig.json --type-check --exclude "__test_data__/*" "src/**/*.ts*"
```

This results in the following error:

`Error File 'C:/path/to/__test_data__/applications.js' is not under 'rootDir' 'C:/path/to/src'. 'rootDir' is expected to contain all source files.`

If I remove one of the underscores from either side of the directory name (e.g. `_test_data_`), the command runs without an error.

#### Expected behavior

The command should run the same regardless of how many underscores are in a directory name.

**Comment by adidahiya** [comment]
Note that the files list is being ignored when you use `--project` (bug, see https://github.com/palantir/tslint/issues/1603). Do you still have this issue when you omit `--project` and `--type-check`?

**Comment by ajafff** [comment]
The error comes from typescript and is only shown, because you enabled `--type-check` on CLI.

If you think this error is a bug, you should report it in the typescript repository after reproducing without tslint.

> Note that the files list is being ignored when you use --project (bug, see #1603)

That's not entirely true. The file list and excludes are taken into account when linting. Only the type checking errors from `--type-check` are not filtered accordingly.
